proto/redpanda/api/dataplane/v1alpha1/secret.proto

```diff
@@ -0,0 +1,191 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+// Secret defienes the secret resource
+message Secret {
+  string id = 1 [
+    (google.api.field_behavior) = IMMUTABLE,
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+  map<string, string> labels = 2 [
+    (buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      },
+    },
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+}
+
+message SecretInput {
+  string id = 1 [(google.api.field_behavior) = IMMUTABLE];
+  map<string, string> labels = 2 [
+    (buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      },
+    },
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+  repeated bytes secret_data = 3 [
+    (google.api.field_behavior) = INPUT_ONLY,
+    (google.api.field_behavior) = REQUIRED
+  ];
```

Comment:
if this is required, should there be buf validation `min_len` or something?

Reply:
Same as above, the protos for anything but user.proto are just drafts for the OpenAPI spec.